src/pages/project/components/EnvironmentModal/index.tsx

```diff
@@ -69,34 +69,56 @@ const EnvironmentModal = (props: IProps) => {
     setValue('key', environmentInfo.key);
   }, [environmentInfo, setValue]);
 
-  const { check, setSearchTime } = useSearchTime();
+  const creatRequestTimeCheck = useRequestTimeCheck();
 
-  const debounceSearch = useMemo(() => {
-    return debounce(
-      async (type: string, value: string) => {
-        const time = Date.now();
-        setSearchTime(time);
-        const res = await checkEnvironmentExist(projectKey, {
-          type,
-          value
+  const debounceNameExist = useMemo(() => {
+    return debounce(async (type:string, value: string) => {
+      const check = creatRequestTimeCheck("name");
```

Comment:
if it is necessary to make different keys for projectKey、toggleKey、environmentKey and segmentKey?

Reply:
This hook will be called in different components, each call will return a new creatRequestTimeCheck, they will not conflict with each other.